### PR TITLE
added GuiObject

### PR DESCRIPTION
**Pull request by aportner**

### CODE REVIEW

File: lib/instances/GuiObject.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	getDefault = function()
+		return false
+	end,
+})
+
+GuiObject.properties.Frame = InstanceProperty.typed("UDim2", {
+	getDefault = function()
+		return UDim2.new()
+	end,
+})
```
Comment (LPGhatguy) [inline lib/instances/GuiObject.lua]: Frame isn't a _property_, is it?

File: lib/instances/GuiObject.lua [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+local Vector2 = import("../types/Vector2")
+local GuiObject = BaseInstance:extend("GuiObject")
+
+GuiObject.properties.Active = InstanceProperty.typed("boolean", {
+	getDefault = function()
+		return true
+	end,
+})
+
+
```
Comment (LPGhatguy) [inline lib/instances/GuiObject.lua]: Extra blank line!

File: lib/instances/GuiObject.lua [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	getDefault = function()
+		return Vector2.new()
+	end,
+})
+
+GuiObject.properties.AutoButtonColor = InstanceProperty.typed("boolean", {
+	getDefault = function()
+		return false
+	end,
+})
```
Comment (LPGhatguy) [inline lib/instances/GuiObject.lua]: We should try to implement the properties in a way that lines up with the real inheritance tree. `AutoButtonColor` is only on `TextButton` and `ImageButton`, and defaults to `true` when inserted via `Instance.new`, which has different defaults than when inserted via Studio.

The best reference for that is probably http://wiki.roblox.com/index.php?title=API:Class/GuiObject

Comment (aportner) [reply lib/instances/GuiObject.lua]: whoops missed this one. was mocking TOO FAST

File: lib/instances/GuiObject.lua (hunk 1 above)
Comment (aportner) [reply lib/instances/GuiObject.lua]: I don't even know where to begin on why I put this here

**Pull request merged**